src/MetaMorpho.sol

```diff
@@ -345,6 +345,7 @@ contract MetaMorpho is ERC4626, ERC20Permit, Ownable2Step, Multicall, IMetaMorph
                 Id id = withdrawQueue[i];
 
                 if (config[id].cap != 0) revert ErrorsLib.InvalidMarketRemovalNonZeroCap(id);
+                if (pendingCap[id].validAt != 0) revert ErrorsLib.PendingCap(id);
```

Comment:
Why the revert for this condition?

Reply:
There are 2 ways to remove a markets: 
1. Either there's no supply on it, the cap has been set to 0 previously and the market can be safely removed by triggering `updateWithdrawQueue`.
2. Either there's some supply on it. In this case, the cap must be set to 0, `submitMarketRemoval` must be triggered. After the timelock has passed `updateWithdrawQueue` can be triggered.

Reply:
So this check make sure that no pending cap has been set to a positive value while the cap is now at 0. It enforces to revoke the pending value to make sure the state is clean and no market will be readded right after being removed.